Minutes from CG plenary 2024-02-28 (#630)
solid/specification: meetings/2024-02-28.md

# W3C Solid Community Group: 28 Feb 2024 plenary

* Date: 2024-02-28T14:00:00Z
* Call: https://meet.jit.si/solid-cg
* Chat: https://matrix.to/#/#solid_specification:gitter.im
* Repository: https://github.com/solid/specification

> 

Reviewer: ```suggestion
Status: Published

```

* Join queue to talk.
* Topics can be proposed at the bottom of the agenda to be discussed as time allows. Make it known if a topic is urgent or cannot be postponed.

### Participation and Code of Conduct
* [Join the W3C Solid Community Group](https://www.w3.org/community/solid/join), [W3C Account Request](http://www.w3.org/accounts/request), [W3C Community Contributor License Agreement](https://www.w3.org/community/about/agreements/cla/)
* [Solid Code of Conduct](https://github.com/solid/process/blob/main/code-of-conduct.md), [Positive Work Environment at W3C: Code of Ethics and Professional Conduct](https://www.w3.org/Consortium/cepc/)
* Operating principle for effective participation is to allow access across disabilities, across country borders, and across time. Feedback on tooling and meeting timing is welcome.
* If this is your first time, welcome! please introduce yourself.

### Scribes
* Sarven Capadisli

### Introductions
* JW: In Solid for ~5 years. Worked at Inrupt for the last year and a half; now again doing research in SW/Ethical Web ??? at U. of Oxford.

---

## Topics

### NDNComm panel announcement
URL: https://www.nist.gov/news-events/events/ndncomm2024

> >RG: Prof Lixia Zhang (UCLA) is organizing NDNComm at [NIST](https://www.nist.gov) with a panel titled "From local-first to fully-decentralized applications"
> >RG: Description: There exist many ongoing efforts aiming to steer Internet applications towards decentralized realizations; one such work was presented at last IETF "Local-first software, resilient and secure collaboration".  This panel will showcase examples of decentralized designs; articulate their commonalities and differences; share experience from the different efforts; and identify remaining challenges in building secure, resilient, and easy to use and deploy codebase with well defined API to facilitate new generations of decentralized apps.
> >RG: Would anyone in this meeting be interested in, or know someone who would be interested in, speaking about Solid and, especially, Linked Data, at this Panel? Please get in touch with me over chat. 
> >RG: Everyone is invited to join as participants.

Reviewer 2: ```suggestion
RG: Prof Lixia Zhang (UCLA) is organizing NDNComm at [NIST](https://www.nist.gov) with a panel titled "From local-first to fully-decentralized applications"
RG: Description: There exist many ongoing efforts aiming to steer Internet applications towards decentralized realizations; one such work was presented at last IETF — "Local-first software, resilient and secure collaboration".  This panel will showcase examples of decentralized designs; articulate their commonalities and differences; share experience from the different efforts; and identify remaining challenges in building secure, resilient, and easy to use and deploy codebase with well defined API to facilitate new generations of decentralized apps.
RG: Would anyone in this meeting be interested in, or know someone who would be interested in, speaking about Solid and, especially, Linked Data, at this Panel? Please get in touch with me over chat. 
RG: Everyone is invited to join as participants.
```

* ML: Generated by federation server. In the future, the data is modified on an instance, and notifies to federation to regenerate.
* eP: Could it be done using a notification channel?
* ML: Yes.
* eP: Requirements on the source of the data?
* ML: Profiles of users?
* eP: Yes.
* ML: It's RDF. There are not WebID profiles yet.
* eP: How does adding to the index work?
* ML: We index everyone. In the future, don't know. Need to consult. Perhaps consent.
* MdJ: Can you summarise the conclusion on what practitioners need from specs?
* ML: See video from the [presentation on Solid practitioners](https://github.com/solid-contrib/practitioners/blob/main/meetings/2024-02-19.md). Federated or distributed. 
* MdJ: Do you think we need to add this to the Solid spec?
* ML: I don't know. We are in the client-client domain. TypeIndex type of thing, which is not in the Protocol. It can be an addition.
* MdJ: If you want different servers together, then they need to communicate.
* ML: In the client-client standard we say we use this kind of index.
* TBL: Data serialization of client-client should be in different spec.
* eP: How do you <abbr title="Follow Your Nose">FYN</abbr>?

> * ML: We're making SPARQL queries with Comunica. Fetch metadata index, then know where to go for, e.g., skills, cities. Then from the skill index, then continue to find the appropriate index with defined `ex:forProperty`, `ex:forValue`. Trying to make generic vocab for indexing. And also `rdfs:seeAlso`.

Reviewer: ```suggestion
* ML: We're making SPARQL queries with Communica. Fetch metadata index, then know where to go for, e.g., skills, cities. Then from the skill index, then continue to find the appropriate index with defined `ex:forProperty`, `ex:forValue`. Trying to make generic vocab for indexing. And also `rdfs:seeAlso`.
```

Author: It's actually spelt like that: https://comunica.dev/about/

Reviewer 2: It's worth linking.
```suggestion
* ML: We're making SPARQL queries with [Comunica](https://comunica.dev/about/). Fetch metadata index, then know where to go for, e.g., skills, cities. Then from the skill index, then continue to find the appropriate index with defined `ex:forProperty`, `ex:forValue`. Trying to make generic vocab for indexing. And also `rdfs:seeAlso`.
```

* MdJ: For example, in FB, everything is linked to your profile. Now, if you look at requirements for identity systems, it says don't leak global identity. Maybe want to use an app to give an access to bookmarks, but don't want to tell it my WebID.
* SC: Henry Story was also touching on this quite a bit, presenting only the kind of credential that's needed to access something. Selective disclosure. This had to do with readable ACLs, extending WAC so the Controller can make the ACL readable without making it writable.
* SC: And about reusing existing specs (such as OAuth), Solid in itself is not the most adequate place to solve those hard authentication/authorization problems for the web. Try to reuse and extend minimally.
* eP: In some contexts, you don't want to use certain identifiers. In others, need to establish global identity. Many people use same name; they want to establish this kind of name/identity. So be relatable. Both directions are valid for different situations. Applicable to different use cases.
* eP: prior work https://kushaldas.in/posts/solid-project-webid-and-privacy.html

> * TBL: Agree. Need to make sure the Solid community understands. People designing stuff for nuclear warheads.. ??? or someone linked to their online profiles. CC would give a token number, where it would work only once. Could that with WebID. Go to restaurant, happy ot authn as me, but also come up with my phone and have it generate a emphemeral ID where I'd only give my address. For whistleblowers, for example, where they dump some information.

Reviewer: ```suggestion
* TBL: Agree. Need to make sure the Solid community understands. People designing stuff for nuclear warheads.. ??? or someone linked to their online profiles. CC would give a token number, where it would work only once. Could that with WebID. Go to restaurant, happy to authn as me, but also come up with my phone and have it generate a ephemeral ID where I'd only give my address. For whistleblowers, for example, where they dump some information.
```
